queue/priority_queue.go

```diff
@@ -0,0 +1,76 @@
+package queue
+
+import (
+	"container/heap"
+)
+
+// PriorityQueueItem is an interface that represents items in a PriorityQueue.
+// Users of PriorityQueue will need to define a Less function such that
+// PriorityQueue will be able to use that to build and restore an underlying
+// heap.
+type PriorityQueueItem interface {
+	// Less must return true if this item is ordered before other and false
+	// otherwise.
+	Less(other PriorityQueueItem) bool
```

Comment:
nice work on this interface!

Reply:
Thanks :) It's now merged through an other PR, but fixed the comments in my commit to be fully equal.

Reply:
I really thought I took that comment on `Less`. Ok, next time we should split off the common functionality between two prs.